source/grid/tria_accessor.cc

```diff
@@ -1223,6 +1223,139 @@ TriaAccessor<structdim, dim, spacedim>::intermediate_point (const Point<structdi
+   * b = Y*Kb.
+   */
+  template <int dim>
+  struct TransformR2UAffine
+  {
+    static const double KA[GeometryInfo<dim>::vertices_per_cell][dim];
+    static const double Kb[GeometryInfo<dim>::vertices_per_cell];
+  };
+
+
+  /*
+    Octave code:
+    M=[0 1; 1 1];
+    K1 = transpose(M) * inverse (M*transpose(M));
+    printf ("{%f, %f},\n", K1' );
+  */
+  template <>
+  const double
+  TransformR2UAffine<1>::
+  KA[GeometryInfo<1>::vertices_per_cell][1] =
+  {
+    {-1.000000},
+    {1.000000}
+  };
+
+  template <>
+  const double
+  TransformR2UAffine<1>::
+  Kb[GeometryInfo<1>::vertices_per_cell] = {1.000000, 0.000000};
+
+
+  /*
+    Octave code:
+    M=[0 1 0 1;0 0 1 1;1 1 1 1];
+    K2 = transpose(M) * inverse (M*transpose(M));
+    printf ("{%f, %f, %f},\n", K2' );
+  */
+  template <>
+  const double
+  TransformR2UAffine<2>::
+  KA[GeometryInfo<2>::vertices_per_cell][2] =
+  {
+    {-0.500000, -0.500000},
+    { 0.500000, -0.500000},
+    {-0.500000,  0.500000},
+    { 0.500000,  0.500000}
+  };
+
+  /*
+    Octave code:
+    M=[0 1 0 1 0 1 0 1;0 0 1 1 0 0 1 1; 0 0 0 0 1 1 1 1; 1 1 1 1 1 1 1 1];
+    K3 = transpose(M) * inverse (M*transpose(M))
+    printf ("{%f, %f, %f, %f},\n", K3' );
+  */
+  template <>
+  const double
+  TransformR2UAffine<2>::
+  Kb[GeometryInfo<2>::vertices_per_cell] =
+  {0.750000,0.250000,0.250000,-0.250000 };
+
+
+  template <>
+  const double
+  TransformR2UAffine<3>::
+  KA[GeometryInfo<3>::vertices_per_cell][3] =
+  {
+    {-0.250000, -0.250000, -0.250000},
+    { 0.250000, -0.250000, -0.250000},
+    {-0.250000,  0.250000, -0.250000},
+    { 0.250000,  0.250000, -0.250000},
+    {-0.250000, -0.250000,  0.250000},
+    { 0.250000, -0.250000,  0.250000},
+    {-0.250000,  0.250000,  0.250000},
+    { 0.250000,  0.250000,  0.250000}
+
+  };
+
+
+  template <>
+  const double
+  TransformR2UAffine<3>::
+  Kb[GeometryInfo<3>::vertices_per_cell] =
+  {0.500000,0.250000,0.250000,0.000000,0.250000,0.000000,0.000000,-0.250000};
+}
+
+
+template <int structdim, int dim, int spacedim>
+Point<structdim>
+TriaAccessor<structdim, dim, spacedim>
+::real_to_unit_cell_affine_approximation (const Point<spacedim> &point) const
+{
+  // A = vertex * KA
+  DerivativeForm<1,structdim,spacedim> A;
+
+  // copy vertices to avoid expensive resolution of vertex index inside loop
+  std::array<Point<spacedim>, GeometryInfo<structdim>::vertices_per_cell> vertices;
+  for (unsigned int v=0; v<GeometryInfo<structdim>::vertices_per_cell; ++v)
+    vertices[v] = this->vertex(v);
+  for (unsigned int d=0; d<spacedim; ++d)
+    for (unsigned int v=0; v<GeometryInfo<structdim>::vertices_per_cell; ++v)
+      for (unsigned int e=0; e<structdim; ++e)
+        A[d][e] += vertices[v][d] * TransformR2UAffine<structdim>::KA[v][e];
+
+  // b = vertex * Kb
+  Tensor<1,spacedim> b = point;
+  for (unsigned int v=0; v<GeometryInfo<structdim>::vertices_per_cell; ++v)
+    b -= this->vertex(v) * TransformR2UAffine<structdim>::Kb[v];
+
```

Comment:
An alternative to transforming a single point according to this mapping would be to let this function return `A` and `b`, i.e., the affine approximation to the cell. It would then be a two-liner at the application site to compute the real-to-unit approximation with the affine method. It would be more flexible - on the other hand I don't really see an application of the affine approximation except for this single purpose. What do you guys think?